server/models.py

```diff
@@ -0,0 +1,581 @@
+    districts = relationship(
+        "District",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="District.name",
+    )
+
+    # Results of the election
+    results = relationship(
+        "ElectionResult",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="ElectionResult.id",
+    )
+
+    # When a user deletes an election, we keep it in the database just in case
+    # they change their mind, but flag it so that we can restrict access
+    deleted_at = Column(UTCDateTime)
+
+    @validates('polls_timezone')
+    def convert_upper(self, _key, polls_timezone):
+        return polls_timezone.upper()
+
+    __table_args__ = (UniqueConstraint("organization_id", "election_name"),)
+
+
+class ElectionResultSource(str, enum.Enum):
+    FILE = "File"
+    DATA_ENTRY = "Data Entry"
+
+class ElectionResultStatus(str, enum.Enum):
+    READY="Ready"
+    FAILED="Failed"
+    PUBLISHED="Published"
+
+class ElectionResult(BaseModel):
+    id = Column(String(200), primary_key=True)
+    total_ballots_cast = Column(String(200), nullable=False)
+    source = Column(Enum(ElectionResultSource), nullable=False)
+    status = Column(Enum(ElectionResultStatus), default=ElectionResultStatus.READY, nullable=False)
+
+    #Election to which results belong
+    election_id = Column(
+        String(200),
+        ForeignKey("election.id", ondelete="cascade"),
+        nullable=False
+    )
+    election = relationship("Election", back_populates="results")
+    #Jurisdiction to which results belong
+    jurisdiction_id = Column(
+        String(200),
+        ForeignKey("jurisdiction.id", ondelete="cascade"),
+        nullable=False
+    )
+    jurisdiction = relationship("Jurisdiction")
+    #Precinct to which results belong
+    precinct_id = Column(
+        String(200),
+        ForeignKey("precinct.id", ondelete="cascade"),
+        nullable=False
+    )
+    precinct = relationship("Precinct")
+    # Contests in the election
+    contests = relationship(
+        "Contest",
+        back_populates="election_result",
+        uselist=True,
+        passive_deletes=True,
+        order_by="Contest.name",
+    )
+    # When a user deletes an election result, we keep it in the database just in case
+    # they change their mind, but flag it so that we can restrict access
+    deleted_at = Column(UTCDateTime)
+
+    __table_args__ = (UniqueConstraint("election_id", "jurisdiction_id"),)
+
+
+# these are typically counties
+class Jurisdiction(BaseModel):
+    id = Column(String(200), primary_key=True)
+    name = Column(String(200), nullable=False)
+
+    election_id = Column(
+        String(200),
+        ForeignKey("election.id", ondelete="cascade"),
+        nullable=False
+    )
+    election = relationship("Election", back_populates="jurisdictions")
+
+    __table_args__ = (UniqueConstraint("name", "election_id"),)
```

Comment:
I think better to have this in the `election_id`, `name` order, as that allows the index to be used partially on just `election_id`.